### PR TITLE
feat: git delete branch | FE

### LINKED ISSUE #12612: [Task]: Remove segment header line from git branch list
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

The design system update was taken in #12530.

This task will update the BranchList component to not use the styled rule from ListSegmentHeader component.

### PR DESCRIPTION
**Pull request by IAmAnubhavSaini**
## Description

This PR cleans up git branch popup, and introduces a new feature: delete branch.

Fixes #12612, #12677,  #12074

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How Has This Been Tested?

Tests will be provided.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: feat-git-delete-branch 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 56.34 **(0)** | 37.76 **(0.03)** | 35.89 **(-0.05)** | 56.58 **(0.01)**
 :green_circle: | app/client/src/actions/gitSyncActions.ts | 62.36 **(0.4)** | 100 **(0)** | 3.7 **(-0.38)** | 64.71 **(2.21)**
 :red_circle: | app/client/src/api/GitSyncAPI.tsx | 45.9 **(0.14)** | 16.67 **(0)** | 10.53 **(-0.58)** | 49.09 **(0.03)**
 :red_circle: | app/client/src/ce/constants/messages.ts | 77.19 **(-0.04)** | 100 **(0)** | 32.55 **(-0.13)** | 81.9 **(-0.14)**
 :green_circle: | app/client/src/components/ads/Button.tsx | 90.97 **(0)** | 78.26 **(2.61)** | 95.56 **(0)** | 91.55 **(0)**
 :red_circle: | app/client/src/pages/Editor/gitSync/utils.ts | 95.45 **(-4.55)** | 66.67 **(-33.33)** | 83.33 **(-16.67)** | 93.33 **(-6.67)**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/components/BetaTag.tsx** | **100** | **100** | **100** | **100**
 :green_circle: | app/client/src/pages/Editor/gitSync/components/BranchList.tsx | 29.31 **(5.82)** | 19.51 **(9.38)** | 0 **(-2.5)** | 32.08 **(6.58)**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/components/BranchListItem.tsx** | **42.31** | **0** | **0** | **44**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/components/BranchListItemContainer.tsx** | **100** | **83.33** | **100** | **100**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/components/BranchMoreMenu.tsx** | **37.5** | **0** | **0** | **41.38**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/components/DangerMenuItem.tsx** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/components/DefaultTag.tsx** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/components/LocalBranchList.tsx** | **58.33** | **66.67** | **25** | **70**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/components/RemoteBranchList.tsx** | **87.5** | **83.33** | **66.67** | **87.5**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/components/RemoteBranchListItem.tsx** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/hooks/index.ts** | **100** | **100** | **0** | **100**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/hooks/useActiveHoverIndex.ts** | **15.79** | **0** | **0** | **14.29**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/hooks/useFilteredBranches.ts** | **18.75** | **0** | **0** | **15.38**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/hooks/useGitConnect.ts** | **29.41** | **100** | **0** | **28.57**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/hooks/useHover.ts** | **11.11** | **0** | **0** | **14.29**
 :sparkles: :new: | **app/client/src/pages/Editor/gitSync/hooks/useSSHKeyPair.ts** | **16.67** | **0** | **0** | **18.52**
 :red_circle: | app/client/src/reducers/uiReducers/gitSyncReducer.ts | 10.94 **(-0.73)** | 0 **(0)** | 1.75 **(-0.14)** | 10.94 **(-0.73)**
 :red_circle: | app/client/src/sagas/GitSyncSagas.ts | 9.97 **(-0.16)** | 0.96 **(-0.02)** | 8.33 **(-0.37)** | 12.17 **(-0.13)**
 :green_circle: | app/client/src/selectors/commentsSelectors.ts | 85.25 **(1.64)** | 64.71 **(2.95)** | 73.33 **(0)** | 90.59 **(2.35)**
 :red_circle: | app/client/src/selectors/gitSyncSelectors.tsx | 71.05 **(-0.09)** | 12.64 **(0)** | 14 **(-0.29)** | 62.11 **(-0.93)**</details>

### CODE REVIEW

**Comment by IAmAnubhavSaini** [comment]
**Issue while installing packages**

Screenshot 2022-04-11 at 6 26 35 AM

**Comment by IAmAnubhavSaini** [comment]
https://user-images.githubusercontent.com/1573771/163066371-a8e7fb9d-eaf3-4e37-be6e-22e600fdfe9f.mov

Screenshot 2022-04-13 at 4 10 10 AM
Screenshot 2022-04-13 at 4 10 20 AM
Screenshot 2022-04-13 at 4 10 26 AM

File: app/client/src/pages/Editor/gitSync/components/BetaTag.test.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    expect(actual?.innerHTML.includes("BETA")).toBeTruthy();
+
+    // styles
+    expect(actual).toHaveStyleRule("height", "16px");
+    expect(actual).toHaveStyleRule("width", "48px");
+    expect(actual).toHaveStyleRule("display", "flex");
+    expect(actual).toHaveStyleRule("justify-content", "center");
+    expect(actual).toHaveStyleRule("align-items", "center");
+    expect(actual).toHaveStyleRule("color", "#191919");
+    expect(actual).toHaveStyleRule("border", "1px solid #191919");
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/gitSync/components/BetaTag.test.tsx:24]: Can we use `Colors.COD_GRAY` here?

File: app/client/src/pages/Editor/gitSync/components/BranchList.tsx [hunk 2]
```diff
@@ -66,74 +66,43 @@ const BranchDropdownContainer = styled.div`
...
   display: -webkit-box;
   -webkit-line-clamp: 3;
   -webkit-box-orient: vertical;
   overflow: hidden;
+
   & ${IconWrapper} {
     display: inline;
   }
+
   & span {
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/gitSync/components/BranchList.tsx:83]: since span is changed to div below, the identifier here should also change.

**Review by ankitakinger — CHANGES_REQUESTED** [review]
Will this file be re-written?
app/client/src/pages/Editor/gitSync/components/BranchList.test.tsx

File: app/client/src/pages/Editor/gitSync/hooks/useSSHKeyPair.ts [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 import {
   getSSHKeyDeployDocUrl,
   getSshKeyPair,
-} from "selectors/gitSyncSelectors";
+} from "../../../../selectors/gitSyncSelectors";
+import { useCallback, useEffect, useState } from "react";
+import {
+  generateSSHKeyPair,
+  getSSHKeyPair,
+} from "../../../../actions/gitSyncActions";
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/gitSync/hooks/useSSHKeyPair.ts:10]: can use `actions/gitSyncActions` and `selectors/gitSyncSelectors` instead?

File: app/client/src/pages/Editor/gitSync/components/BetaTag.test.tsx (hunk 1 above)
Comment (IAmAnubhavSaini) [reply app/client/src/pages/Editor/gitSync/components/BetaTag.test.tsx:24]: I am testing for exact value.

File: app/client/src/pages/Editor/gitSync/hooks/useSSHKeyPair.ts (hunk 3 above)
Comment (IAmAnubhavSaini) [reply app/client/src/pages/Editor/gitSync/hooks/useSSHKeyPair.ts:10]: I have already updated similar code in other places too. Need to push that commit.

**Comment by IAmAnubhavSaini** [comment]
> Will this file be re-written? app/client/src/pages/Editor/gitSync/components/BranchList.test.tsx

Check utils test file. The function moved and so did the tests.

**Review by ankitakinger — APPROVED** [review]
LGTM

**Pull request merged**